stl/inc/optional

```diff
@@ -53,14 +53,17 @@ protected:
     _THROW(bad_optional_access{});
 }
 
-struct _Nontrivial_Dummy_type {
-    constexpr _Nontrivial_Dummy_type() noexcept {}
+struct _Nontrivial_dummy_type {
+    constexpr _Nontrivial_dummy_type() noexcept {
+        // This default constructor is user-provided to avoid zero-initialization when objects are value-initialized.
+    }
 };
+_STL_INTERNAL_STATIC_ASSERT(!is_trivially_default_constructible_v<_Nontrivial_dummy_type>);
 
 template <class _Ty, bool = is_trivially_destructible_v<_Ty>>
 struct _Optional_destruct_base { // either contains a value of _Ty or is empty (trivial destructor)
     union {
-        _Nontrivial_Dummy_type _Dummy;

```

Comment:
I guess that revert was meant to change the name to `_Nontrivial_dummy_type `

Reply:
I broke this generating code for the "with just the change to `optional`'s default constructor" bit in https://github.com/microsoft/STL/pull/878#issuecomment-639696118. Thanks for catching it.